pkg/resources/diff_suppressions_test.go

```diff
@@ -80,3 +80,73 @@ func Test_IgnoreAfterCreation(t *testing.T) {
 		assert.True(t, result)
 	})
 }
+
+func Test_NormalizeAndCompareIdentifiersSet(t *testing.T) {
+	rawDataWithValues := func(values []any) *schema.ResourceData {
+		return schema.TestResourceDataRaw(t, map[string]*schema.Schema{
+			"value": {
+				Required: true,
+				Type:     schema.TypeSet,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		}, map[string]any{
+			"value": values,
+		})
+	}
+	emptyResourceData := rawDataWithValues([]any{})
+
+	t.Run("validation: size key", func(t *testing.T) {
+		assert.False(t, resources.NormalizeAndCompareIdentifiersInSet("value")("value.#", "1", "2", emptyResourceData))
+	})
+
+	t.Run("validation: case mismatch", func(t *testing.T) {
+		resourceData := rawDataWithValues([]any{"SCHEMA.OBJECT.IDENTIFIER"})
+		assert.False(t, resources.NormalizeAndCompareIdentifiersInSet("value")("value.doesnt_matter", "schema.object.identifier", "", resourceData))
+		// TODO: Cannot be tested with schema.TestResourceDataRaw because it doesn't populate raw state which is used in the cases below
+		// assert.False(t, resources.NormalizeAndCompareIdentifiersInSet("value")("value.doesnt_matter", "", `"schema"."object"."identifier"`, resourceData))
+	})
+
+	t.Run("validation: case mismatch quoted identifier in the state", func(t *testing.T) {
+		resourceData := rawDataWithValues([]any{`"SCHEMA"."OBJECT"."IDENTIFIER"`})
+		assert.False(t, resources.NormalizeAndCompareIdentifiersInSet("value")("value.doesnt_matter", "schema.object.identifier", "", resourceData))
+		// TODO: Cannot be tested with schema.TestResourceDataRaw because it doesn't populate raw state which is used in the cases below
+		// assert.False(t, resources.NormalizeAndCompareIdentifiersInSet("value")("value.doesnt_matter", "", `"schema"."object"."identifier"`, resourceData))
+	})
+
+	t.Run(`change detected from schema.object.identifier to "schema"."object"."identifier" with schema.object.identifier in state`, func(t *testing.T) {
```

Comment:
hmm, I don't get this test, they both have exactly the same values on not the quoted parts vs unquoted. What am I missing here?

Reply:
Not sure what you mean here. They're the same because we only care in those cases about quoting, not casing (here we ignore that in Snowflake id = "ID"). That's what we talked about we don't care that much about casing because we always quote identifiers we get from the config and this should be clearly communicated. The diff suppress leverages that assumption and doesn't care about original quoting because we would have to hold that information somehow (e.g. that 1 and 2 parts were quoted, but part 3 wasn't). We can leave it as a future improvement, but for now, this should work correctly.

Reply:
The test name was misleading (detected vs suppressed), so I did not understand why we are expecting a change - now, it's clear.